### PR TITLE
Remove injected_sql. Store non-ephemeral injected_sql in compiled_sql

### LINKED ISSUE #2762: Remove injected_sql
### Describe the feature
Instead of adding `injected_sql` to the model during compilation, dbt should overwrite `compiled_sql`. There will probably be a lot of test fallout!

Split from #2671

### PR DESCRIPTION
**Pull request by gshank**
resolves #2762

### Description

The 'injected_sql' field in the manifest nodes is confusing, since it's another form of compiled_sql. The injected_sql field is used internally in compiling ephemeral nodes. This code removed injected_sql from the classes and artifacts, and copies it into 'compiled_sql' for executable nodes. Ephemeral nodes will retain their original 'compiled_sql', since it didn't work to overwrite compiled_sql with injected_sql for those nodes. Since ephemeral nodes do not show up in executable SQL files, this will not hopefully be an issue.

### Checklist
 - [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
 - [x] I have run this code in development and it appears to resolve the stated issue
 - [x] This PR includes tests, or tests are not required/relevant for this PR
 - [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt next" section.

### CODE REVIEW

File: test/integration/004_simple_snapshot_test/test_simple_snapshot.py [hunk 1]
```diff
@@ -919,5 +919,7 @@ def _snapshot_and_assert_revived(self):
         for result in revived_records:
             # result is a tuple, the dbt_valid_from is second and dbt_valid_to is latest
             self.assertIsInstance(result[1], datetime)
-            self.assertGreaterEqual(result[1].replace(tzinfo=pytz.UTC), self._invalidated_snapshot_datetime)
+            # there are milliseconds (part of microseconds in datetime objects) in the
+            # invalidated_snapshot_datetime and not in result datetime so set the microseconds to 0
+            self.assertGreaterEqual(result[1].replace(tzinfo=pytz.UTC), self._invalidated_snapshot_datetime.replace(microsecond=0))
```
Comment (jtcohen6) [inline test/integration/004_simple_snapshot_test/test_simple_snapshot.py:924]: nice catch

**Review by jtcohen6 — APPROVED** [review]
I took this for a spin locally and tried a bunch of hacky things with ephemeral models and dynamic templating (via `run_query`). The `compiled_sql` in the manifest looks good every time.

After we merge this and before a v0.19 prerelease, we'll want to merge the corresponding dbt-docs PR: https://github.com/fishtown-analytics/dbt-docs/pull/148

**Review by kwigley — APPROVED** [review]
Thanks for being patient with me! This looks great, just one small stylistic question

File: core/dbt/context/providers.py [hunk 2]
```diff
@@ -1217,7 +1217,9 @@ def post_hooks(self) -> List[Dict[str, Any]]:
 
     @contextproperty
     def sql(self) -> Optional[str]:
-        return getattr(self.model, 'injected_sql', None)
+        if getattr(self.model, 'extra_ctes_injected', None):
```
[the remark this answers is not in the document]
Comment (gshank) [reply core/dbt/context/providers.py:1220]: I think sometimes that ModelContext is constructed from models that don't have the 'extra_ctes_injected' (and 'injected_sql' too) which is why they were using the 'getattr' function.

**Pull request merged**